source/Irrlicht/CFileSystem.cpp

```diff
@@ -861,6 +859,9 @@ IFileList* CFileSystem::createFileList()
 			}
 			closedir(dirHandle);
 		}
+		#else
+			// no native filesystem enabled
+			return nullptr;

```

Comment:
I don't think burying the return deep inside the function helps readability here

Reply:
That is a point. Maybe that one is better left how it was; do you want me to undo it?

Reply:
I'd prefer that yes